src/_pytest/compat.py

```diff
@@ -354,3 +354,10 @@ def funcargnames(self):
 
     def overload(f):  # noqa: F811
         return f
+
+
+attrs_has_eq = getattr(attr, "__version_info__", (0, 0)) >= (19, 2)
```

Comment:
if we declare the attribute name here, then it can also be used in a `getattr` in the other loop

Reply:
I'm sorry but I don't follow. There's only one getattr and zero loops in this PR? Or do you want me to define attrsfieldhaseq here too?

Reply:
i would like to see a `ATTRS_EQ_FIELD = "eq" if attrs_has_eq else "cmp"` and usage of that constant downstream

Reply:
I agree, we ideally should make all "ifs" for compatilibty here in this module, including the `attrsfieldhaseq` function, although I suspect @RonnyPfannschmidt is suggesting to get rid of the function and having only `ATTRS_EQ_FIELD` (@RonnyPfannschmidt please correct me if I'm wrong).